File: Models/Simplace_Soil_Temperature/src/cpp/Simplace_Soil_Temperature/SoilTemperatureAuxiliary.cpp
#include "SoilTemperatureAuxiliary.h"
using namespace Simplace_Soil_Temperature;

SoilTemperatureAuxiliary::SoilTemperatureAuxiliary() {}

std::vector<double> & SoilTemperatureAuxiliary::getSoilTempArray() { return this->SoilTempArray; }
std::vector<double> & SoilTemperatureAuxiliary::getiSoilTempArray() { return this->iSoilTempArray; }
double SoilTemperatureAuxiliary::getSnowIsolationIndex() { return this->SnowIsolationIndex; }
double SoilTemperatureAuxiliary::getiSoilSurfaceTemperature() { return this->iSoilSurfaceTemperature; }

void SoilTemperatureAuxiliary::setSoilTempArray(std::vector<double> const &_SoilTempArray){
    this->SoilTempArray = _SoilTempArray;
}
void SoilTemperatureAuxiliary::setiSoilTempArray(std::vector<double> const &_iSoilTempArray){
    this->iSoilTempArray = _iSoilTempArray;
}
void SoilTemperatureAuxiliary::setSnowIsolationIndex(double _SnowIsolationIndex) { this->SnowIsolationIndex = _SnowIsolationIndex; }
void SoilTemperatureAuxiliary::setiSoilSurfaceTemperature(double _iSoilSurfaceTemperature) { this->iSoilSurfaceTemperature = _iSoilSurfaceTemperature; }

File: Models/Simplace_Soil_Temperature/src/cs/Simplace_Soil_Temperature/SoilTemperatureAuxiliary.cs
using System;
using System.Collections.Generic;

public class SoilTemperatureAuxiliary 
{
    private double[] _SoilTempArray;
    private double[] _iSoilTempArray;
    private double _SnowIsolationIndex;
    private double _iSoilSurfaceTemperature;
    
    /// <summary>
    /// Constructor of the SoilTemperatureAuxiliary component")
    /// </summary>  
    public SoilTemperatureAuxiliary() { }
    
    
    public SoilTemperatureAuxiliary(SoilTemperatureAuxiliary toCopy, bool copyAll) // copy constructor 
    {
        if (copyAll)
        {
    
            SoilTempArray = new double[toCopy.SoilTempArray.Length];
            for (int i = 0; i < toCopy.SoilTempArray.Length; i++)
                { SoilTempArray[i] = toCopy.SoilTempArray[i]; }
    
            iSoilTempArray = new double[toCopy.iSoilTempArray.Length];
            for (int i = 0; i < toCopy.iSoilTempArray.Length; i++)
                { iSoilTempArray[i] = toCopy.iSoilTempArray[i]; }
    
            SnowIsolationIndex = toCopy.SnowIsolationIndex;
            iSoilSurfaceTemperature = toCopy.iSoilSurfaceTemperature;
        }
    }
    public double[] SoilTempArray
    {
        get { return this._SoilTempArray; }
        set { this._SoilTempArray= value; } 
    }
    public double[] iSoilTempArray
    {
        get { return this._iSoilTempArray; }
        set { this._iSoilTempArray= value; } 
    }
    public double SnowIsolationIndex
    {
        get { return this._SnowIsolationIndex; }
        set { this._SnowIsolationIndex= value; } 
    }
    public double iSoilSurfaceTemperature
    {
        get { return this._iSoilSurfaceTemperature; }
        set { this._iSoilSurfaceTemperature= value; } 
    }
}